src/panels/lovelace/editor/unused-entities/hui-unused-entities.ts

```diff
@@ -163,16 +164,87 @@ export class HuiUnusedEntities extends LitElement {
 
   private _filterUnusedEntities(): void {
     this._filteredUnusedEntities = this._unusedEntities.filter((entity) => {
-      const stateObj = entity ? this.hass!.states[entity] : undefined;
+      const stateObj = this.hass!.states[entity];
       return (
         entity.includes(this._filter) ||
-        computeStateName(stateObj!).includes(this._filter)
+        computeStateName(stateObj).includes(this._filter)
       );
     });
   }
 
   private _sortUnusedEntities(): void {
-    return;
+    this._sortedUnusedEntities = [...this._filteredUnusedEntities];
+    if (this._sortDirection) {
+      this._sortedUnusedEntities.sort((a, b) => {
+        let sort = 0;
+        switch (this._sortColumn) {
+          case 0:
+            sort = this._sortByEntityName(a, b);
+            break;
+          case 1:
+            sort = this._sortByEntityId(a, b);
+            break;
+          case 2:
+            sort = this._sortByDomain(a, b);
+            break;
+          case 3:
+            sort = this._sortByLastChanged(a, b);
+            break;
+          default:
+            return 0;
+        }
+        if (this._sortDirection === "desc") {
+          sort = sort * -1;
+        }
+        return sort;
+      });
+    }
+  }
+
+  private _sortByDomain(a, b): number {
+    const domainA = computeDomain(a).toUpperCase();
+    const domainB = computeDomain(b).toUpperCase();
+    if (domainA < domainB) {
+      return -1;
+    }
+    if (domainA > domainB) {
+      return 1;
+    }
+    return 0;
+  }
+
+  private _sortByEntityId(a, b): number {
+    if (a < b) {
+      return -1;
+    }
+    if (a > b) {
+      return 1;
+    }
+    return 0;
+  }
+
+  private _sortByEntityName(a, b): number {
+    const nameA = computeStateName(this.hass!.states[a]).toUpperCase();
+    const nameB = computeStateName(this.hass!.states[b]).toUpperCase();
+    if (nameA < nameB) {
+      return -1;
+    }
+    if (nameA > nameB) {
+      return 1;
+    }
+    return 0;
+  }
+
+  private _sortByLastChanged(a, b): number {
```

Comment:
If you have a 100 entities, and assuming we're using quick sort, we need to do `n * log(n)` comparisons on average. On each of those we're parsing a date twice. 

Can we convert the entities to a structure that allows us to generate the sorting key for each entity once, and then sort them? We should be able to reuse that structure for each sorting algorithm.

Reply:
Yeah it is on my todo